### PR TITLE
Added Alter Room Code For Room Socket

### PR DESCRIPTION
**Pull request by ryanpolaski**

### CODE REVIEW

File: assets/php/classes/websockets/RoomSocketServer.php [hunk 1]
```diff
@@ -169,7 +169,40 @@ protected function on_client_namechange($user_socket, $message, Room &$room, Acc
 
     protected function on_alter_roomcode($user_socket, $message, Room &$room, Account &$account)
     {
-        // TODO: Implement on_alter_roomcode() method.
+        $roomid = $room->getRoomID();
+        if($room->getRoomCodes() > 0) { //check to see if there is a room code
```
Comment (IsaacHarries) [inline assets/php/classes/websockets/RoomSocketServer.php]: if(count($room->getRoomCodes()) > 0) {

File: assets/php/classes/websockets/RoomSocketServer.php [hunk 2]
```diff
@@ -169,7 +169,40 @@ protected function on_client_namechange($user_socket, $message, Room &$room, Acc
...
+                "Room Code Changed",
+                [
+                    "Code" => $newRoomCode
+                ]
+            );
+            foreach ($this->_clients[$roomid] as $k => $participant) {
+                $this->send($participant, $response);
+            }
+            $this->_clients[$roomid][$newRoomCode] = $user_socket;  //add the new user to the array
+            //generate message
```
Comment (IsaacHarries) [inline assets/php/classes/websockets/RoomSocketServer.php]: remove unnecessary comment

File: assets/php/classes/websockets/RoomSocketServer.php [hunk 3]
```diff
@@ -169,7 +169,40 @@ protected function on_client_namechange($user_socket, $message, Room &$room, Acc
...
+            $response = $this->create_response(
+                "Room Code Changed",
+                [
+                    "Code" => $newRoomCode
+                ]
+            );
+            foreach ($this->_clients[$roomid] as $k => $participant) {
+                $this->send($participant, $response);
+            }
+            $this->_clients[$roomid][$newRoomCode] = $user_socket;  //add the new user to the array
```
Comment (3jackdaws) [inline assets/php/classes/websockets/RoomSocketServer.php]: What does this do?

**Review by 3jackdaws — APPROVED** [review]
fgjabhnslkgjb

**Pull request merged**